stream/classes/hardware/architecture/noc/communication_link.py

```diff
@@ -6,9 +6,7 @@
 class CommunicationLink:
     """Represents a fixed-bandwidth communication link used to communicate between two cores."""
 
-    def __init__(
-        self, sender, receiver, bandwidth, unit_energy_cost, bidirectional=False
-    ) -> None:
+    def __init__(self, sender, receiver, bandwidth, unit_energy_cost, bidirectional=False) -> None:
```

Comment:
Saw that in many places that the argument type is being introduced. Was this supposed to happen here too?

probably could be something along the lines of
`def __init__(self, sender: int | Literal["Any"], receiver: int | Literal["Any"], bandwidth: int, unit_energy_cost: int, bidirectional: bool=False)`

Reply:
Adding type hints is a WIP and I wasn't able to do it everywhere in Stream. They can definitely be added to other classes and functions in the future.

Not sure what the point is of adding `Literal["Any"]` as a type hint? Won't this accept any argument?

The main benefit of type hinting the whole code base is that you can run pytest with the typeguard plugin, which dynamically type checks all function calls.

Reply:
Ahh I see. I think `Core` and `int` are both used throughout the code.